server/app/services/migration/ProgramMigrationService.java

```diff
@@ -105,32 +119,81 @@ public ImmutableMap<String, QuestionDefinition> maybeOverwriteQuestionName(
   }
 
   /**
-   * Generate a new admin name for questions of the format "orginal admin name-n" where "n" is the
-   * next consecutive number such that we don't already have a question with that admin name saved.
-   * For example if the admin name is "sample question" and we already have "sample question-1" and
-   * "sample question-2" saved in the db, the generated name will be "sample question-3"
+   * Generate a new admin name for questions of the format "orginal admin name - a" where "a" is the
+   * next consecutive letter such that we don't already have a question with that admin name saved.
+   * For example if the admin name is "sample question" and we already have "sample question - a"
+   * and "sample question - b" saved in the db, the generated name will be "sample question - c"
    */
-  public String maybeGenerateNewAdminName(String adminName) {
-    // If the question name contains a suffix of the form "-n" (for example "admin-name-1"), we want
-    // to strip off the "-n" before searching for the admin name to ensure all similar names are
-    // returned. This also allows us to correctly increment the suffix of the base admin name so we
-    // don't end up with admin names like "admin-name-1-1".
-    Pattern HYPHEN_DIGIT_PATTERN = Pattern.compile("-[0-9]+");
-    Matcher matcher = HYPHEN_DIGIT_PATTERN.matcher(adminName);
+  String findUniqueAdminName(String adminName, List<String> newNamesSoFar) {
+    if (!nameHasConflict(adminName, newNamesSoFar)) {
+      return adminName;
+    }
+
+    // If the question name contains a suffix of the form " - a" (for example "admin name - a"), we
+    // want
+    // to strip off the " - n" to find the base name. This also allows us to correctly increment the
+    // suffix of the base admin name so we don't end up with admin names like "admin name - a - a".
+    Matcher matcher = SUFFIX_PATTERN.matcher(adminName);
```

Comment:
**question:** Is it possible for something like "Date of birth - parent" or something to get matched? Could we add a test for that use case?

Reply:
Discussed in slack, updated to use ` -_- ` as the delimiter.